src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderInput.tsx

```diff
@@ -278,18 +283,46 @@ export const LimitOrderInput = ({
       response: quoteResponse,
     })
 
-    history.push(LimitOrderRoutePaths.Confirm)
+    const { assetReference, chainId } = fromAssetId(limitOrderQuoteParams.sellAssetId)
+
+    // Trigger loading state while we check the allowance
+    setIsCheckingAllowance(true)
```

Comment:
suggestion (for cleanup in a follow-up): could use `allowanceCryptoBaseUnit` react-query here to avoid setting loading states as part of async flow and ensure that it never automagically runs but rather fetch it with `refetch`.

This would syntactically look very similar, except we would use `isFetching` / `isRefetching` instead of state fields, and would remove the finally block (automagically handled by react-query). Could look something like this (types are off in this example):


```typescript
  const { queryKey, queryFn } = useMemo(() => {
    if (limitOrderQuoteParams === rtkSkipToken)
      return {
        queryKey: ['allowanceCryptoBaseUnit', undefined, COW_SWAP_VAULT_RELAYER_ADDRESS, undefined],
        queryFn: reactQuerySkipToken,
      }

    return reactQueries.common.allowanceCryptoBaseUnit(
      limitOrderQuoteParams.sellAssetId,
      COW_SWAP_VAULT_RELAYER_ADDRESS,
      limitOrderQuoteParams.sellAccountAddress as Address,
    )
  }, [limitOrderQuoteParams])

  const { isRefetching: isRefetchingAllowance, refetch: refetchAllowance } = useQuery({
    queryKey,
    queryFn,
    // Don't automatically run, refetch only
    enabled: false,
    select: selectAllowanceCryptoBaseUnit,
  })

  const onSubmit = useCallback(async () => {
    // No preview happening if wallet isn't connected i.e is using the demo wallet
    if (!isConnected || isDemoWallet) {
      return handleConnect()
    }

    // RTK query returns stale data when `skipToken` is used, so we need to handle that case here.
    // This should never happen because we're meant to disable the confirmation button when this is
    // true, but just in case.
    if (!quoteResponse || limitOrderQuoteParams === rtkSkipToken || !sellAccountId) {
      setActiveQuote(undefined)
      return
    }

    // Everything gets bundled up into a canonical set of data for execution. This is to avoid
    // dramas with race conditions, polling, etc. across the input and execution stages of the limit
    // orders feature.
    setActiveQuote({
      params: {
        ...limitOrderQuoteParams,
        validTo: expiryOptionToUnixTimestamp(expiry),
        buyAmountCryptoBaseUnit,
        accountId: sellAccountId,
      },
      response: quoteResponse,
    })

    try {
      // Check the ERC20 token allowance
      const { data: allowanceOnChainCryptoBaseUnit } = await refetchAllowance({
        throwOnError: true,
      })
      // If approval is required, route there
      if (
        bnOrZero(allowanceOnChainCryptoBaseUnit).lt(limitOrderQuoteParams.sellAmountCryptoBaseUnit)
      ) {
        history.push(LimitOrderRoutePaths.AllowanceApproval)
        return
      }

      // Otherwise, proceed with confirmation
      history.push(LimitOrderRoutePaths.Confirm)
      return
    } catch (e) {
      showErrorToast(e)
    }
  }, [
    isConnected,
    isDemoWallet,
    quoteResponse,
    limitOrderQuoteParams,
    sellAccountId,
    setActiveQuote,
    expiry,
    buyAmountCryptoBaseUnit,
    handleConnect,
    refetchAllowance,
    history,
    showErrorToast,
  ])
```